### PR TITLE
SDIT-2148 Refactor audit to allow reuse

### PR DESCRIPTION
**Pull request by andymarke**
* Create subclass NomisEntity for audit columns with same name
* Create function to convert to DTO
* Improve audit person test

TODO Use for all person contact entities

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/alerts/AlertsResource.kt [hunk 1]
```diff
@@ -914,37 +914,6 @@ data class AlertResponse(
   val audit: NomisAudit,
 )
 
-@Schema(description = "The data held in NOMIS the person or system that created this record")
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/alerts/AlertsResource.kt:917]: move to shared packages

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/alerts/AlertsService.kt [hunk 2]
```diff
@@ -99,7 +100,7 @@ class AlertsService(
 
   private fun OffenderAlert.toAlertResponse() = AlertResponse(
     bookingId = id.offenderBooking.bookingId,
-    bookingSequence = id.offenderBooking.bookingSequence!!.toLong(),
+    bookingSequence = id.offenderBooking.bookingSequence.toLong(),
     alertSequence = id.sequence,
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/alerts/AlertsService.kt:104]: compiler warning fix

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/NomisEntity.kt [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import jakarta.persistence.JoinColumn
+import jakarta.persistence.ManyToOne
+import jakarta.persistence.MappedSuperclass
+import org.hibernate.annotations.Generated
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.helper.EntityOpen
+import java.time.LocalDateTime
+
+@MappedSuperclass
+@EntityOpen
+class NomisEntity {
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/NomisEntity.kt:14]: subclass for any entity that wants to return audit data back to DPS

Comment (mikehalmamoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/NomisEntity.kt:14]: Maybe better to have a class name about "Audit". We have loads of NOMIS entities where we're not modelling the audit columns.

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**